ide/csl.api/src/org/netbeans/modules/csl/editor/completion/MethodParamsTipPaintComponent.java

```diff
@@ -51,20 +51,8 @@ public MethodParamsTipPaintComponent(List<List<String>> params, int idx, JTextCo
     }
     
     public @Override void paint(Graphics g) {
-        Object value = (Map) (Toolkit.getDefaultToolkit().getDesktopProperty("awt.font.desktophints")); //NOI18N
-        Map renderingHints = (value instanceof Map) ? (java.util.Map) value : null;
-        if (renderingHints != null && g instanceof Graphics2D) {
-            Graphics2D g2d = (Graphics2D) g;
-            RenderingHints oldHints = g2d.getRenderingHints();
-            g2d.setRenderingHints(renderingHints);
-            try {
-                super.paint(g2d);
-            } finally {
-                g2d.setRenderingHints(oldHints);
-            }
-        } else {
-            super.paint(g);
-        }
+        GraphicsUtils.configureRenderingHints(g);
+        super.paint(g);
```

Comment:
This isn't the same behaviour.  Should it be?  What are the implications of not being?

Another method could be `GraphicsUtils.withRenderingHints(Graphics g, Consumer<Graphics> paint)` that sets and resets around the paint task?

Reply:
Yes, here is one example of a change in behavior. My inclination here is that resetting the hints is unnecessary, and that it would be worth simplifying the code. In the worst case, some other painting code ends up using the leftover hints--but they are the correct hints that should be used throughout the IDE in any case. Any code that wishes to use explicitly different hints had better set their own hints.

Reply:
The other example of a change in behavior is code that previously just took hints from awt.font.desktophints, but which now ends up doing the whole "isGTK() && gtkShouldAntialias()" etc. condition that is found in some existing places. Arguably doing the same in every place is better.

Reply:
I also added the "put(RenderingHints.KEY_ANTIALIASING, RenderingHints.VALUE_ANTIALIAS_ON)" hint, which enables anti-aliasing for non-text things such as diagonal lines and curves. Some existing sites did this already (presumably the ones that actually needed it).